### PR TITLE
DVOP-2253 create_rds_instance

### PR DESCRIPTION
**Pull request by autmwatl**
Resolves DVOP-2253
Resolves DVOP-2265

### CODE REVIEW

File: groups/rds/security_groups.tf [hunk 1]
```diff
@@ -0,0 +1,28 @@
+resource "aws_security_group" "db_security_group" {
+  name        = "${var.environment}-${var.service}-rds"
+  description = "Allow TLS inbound traffic"
+  vpc_id      = data.aws_vpc.placement.id
+
+  ingress {
+    description      = "Ingress from permitted CIDRs"
+    from_port        = 3306
+    to_port          = 3306
+    protocol         = "-1"
```
Comment (Jata2k) [inline groups/rds/security_groups.tf:10]: This should be TCP

File: groups/rds/locals.tf [hunk 2]
```diff
@@ -1,22 +1,40 @@
...
 
-  certificate_arn = lookup(local.secrets, "certificate_arn", null)
-  dns_zone_name = local.secrets.dns_zone_name
+
+  certificate_arn             = lookup(local.secrets, "certificate_arn", null)
+  dns_zone_name               = local.secrets.dns_zone_name
   load_balancer_dns_zone_name = local.secrets.load_balancer_dns_zone_name
-  placement_vpc_pattern = local.secrets.placement_vpc_pattern
+
+  environment                 = local.secrets.environment_name
```
Comment (Jata2k) [inline groups/rds/locals.tf:32]: This can be a variable. No need to add to vault

File: groups/rds/providers.tf [hunk 3]
```diff
@@ -1,3 +1,10 @@
 provider "aws" {
   region = var.region
+  default_tags {
+    tags = {
+      Environment = var.environment
+      Owner       = var.team
```
Comment (Jata2k) [inline groups/rds/providers.tf:6]: I'd change this to `Team` to match the variable name

File: groups/rds/variables.tf [hunk 4]
```diff
@@ -4,6 +4,7 @@ variable "account_name" {
 }
 
 variable "environment" {
+  default     = "devops1"
```
Comment (DrewHarvard) [inline groups/rds/variables.tf:7]: We wouldn't specify a default for `environment`. Let this variable be picked up from the profile.

File: groups/rds/security_groups.tf [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+
+  ingress {
+    description      = "Ingress from permitted CIDRs"
+    from_port        = 3306
+    to_port          = 3306
+    protocol         = "tcp"
+    cidr_blocks      = concat(local.placement_subnet_cidrs, local.automation_subnet_cidrs)
+  }
+
+  egress {
```
Comment (DrewHarvard) [inline groups/rds/security_groups.tf:14]: Can we add a description here please:
`description = "Allow outbound traffic"`

File: groups/rds/variables.tf [hunk 6]
```diff
@@ -42,3 +43,37 @@ variable "team" {
...
+  type        = string
+}
+
+variable "db_engine_version" {
+  default = "8.0"
+}
+
+variable "db_instance_class" {
+  default = "db.t4g.small"
+}
```
Comment (DrewHarvard) [inline groups/rds/variables.tf:59]: Can we add the `type` and a `description` to these variables for consistency please.

File: groups/rds/security_groups.tf [hunk 7]
```diff
@@ -0,0 +1,28 @@
...
+  name        = "${var.environment}-${var.service}-rds"
+  description = "Allow TLS inbound traffic"
+  vpc_id      = data.aws_vpc.placement.id
+
+  ingress {
+    description      = "Ingress from permitted CIDRs"
+    from_port        = 3306
+    to_port          = 3306
+    protocol         = "tcp"
+    cidr_blocks      = concat(local.placement_subnet_cidrs, local.automation_subnet_cidrs)
```
Comment (Jata2k) [inline groups/rds/security_groups.tf:11]: I'd move this concatenation into the locals file. That way we can do all our manipulation in one place and it makes the definition here easier to read. For example [here](https://github.com/companieshouse/kafka-stack/blob/main/groups/kafka/locals.tf#L12-L17) and [here](https://github.com/companieshouse/terraform-modules/blob/main/aws/kafka/security-groups.tf#L11). It also then gives us a common approach for modules

File: groups/rds/rds.tf [hunk 8]
```diff
@@ -0,0 +1,53 @@
...
+  db_subnet_group_name            = aws_db_subnet_group.db_subnet_group.name
+  parameter_group_name            = aws_db_parameter_group.db_parameter_group.name
+  option_group_name               = aws_db_option_group.db_option_group.name
+  storage_type                    = var.db_storage_type
+  port                            = local.db_port
+  enabled_cloudwatch_logs_exports = var.rds_cloudwatch_logs_exports
+  deletion_protection             = true
+  backup_retention_period         = 7
+  backup_window                   = "03:00-06:00"
+  maintenance_window              = "Sat:00:00-Sat:03:00"
```
Comment (DrewHarvard) [inline groups/rds/rds.tf:47]: Can we make these variables for consistency

**Review by DrewHarvard — APPROVED** [review]

**Review by Jata2k — APPROVED** [review]

**Pull request merged**